### ISSUE TITLE
Support pcre modifiers via /pattern/FLAGS syntax

### ISSUE THREAD

**Issue by mqudsi** [martanne/vis#614]
I can't tell if this was discussed, so please forgive me if it was a purposeful omission. However, it would be handy if vis could support the de facto standardized `/pattern/FLAGS` method of setting PCRE modifiers for the search terms. 

For example, unless I'm mistaken, the only way to currently perform a case-insensitive search is to stick the inline pattern modifier `(?i)` at the start of a search pattern. If this were implemented, you could replace `/(?i)foo` with `/foo/i`, which apart from being several characters shorter also doesn't require using the shift key for operations and is compatible with most PCRE interfaces.

`/` is already a reserved character (e.g. `//` does not find a literal `/`) so there is no breakage in doing so.

**Comment by martanne** [comment]
In general I prefer POSIX ERE over PCRE, but I agree that `/foo/i` would at times be convenient.

However, this is incompatible with sam's command language where `i` is already used as an insert command (i.e. `:/pattern/i/text/` is a valid command).

**Comment by mqudsi** [comment]
So if there’s only one unescaped /, can we assume it is followed by modifiers? As in, is `:/search/xxx` a valid sam expression or would it be safe to presume xxx is a flag in this case?

**Comment by martanne** [comment]
The basic syntax is `[range] command` of which `/search/xxx` is an instance which applies `xxx` to the next occurrence of `search`. Hence the question becomes which commands work without an argument:

 * `p` creates a selection, this is the default command and can be omitted
 * `d` deletes the range
 * `|`, `<` and `>` fall back to the last shell command used

Modifiers do not seem to easily fit into this. In the past a global option like `:set ignorecase` was briefly discussed.

**Comment by martanne** [comment]
Depending on the used regex backend (i.e. your libc or a TRE enabled build) you could prepend the option `(?i)` to your regex.

**Comment by mqudsi** [comment]
Yup, I actually mentioned that in my initial report. Thanks, though!
